/* Copyright 2019 Istio Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package stackdriver.config.v1alpha1;

import "google/protobuf/duration.proto";

message PluginConfig {
  // next id: 7

  // Optional. Controls whether to export server access log.
  bool disable_server_access_logging = 1;

  // Optional. FQDN of destination service that the request routed to, e.g.
  // productpage.default.svc.cluster.local. If not provided, request host header
  // will be used instead
  string destination_service_name = 2;

  // Optional. Controls whether or not to export mesh edges to a mesh edges
  // service. This is disabled by default.
  bool enable_mesh_edges_reporting = 3;

  // Optional. Allows configuration of the time between calls out to the mesh
  // edges service to report edges. The minimum configurable duration is `10s`.
  // The default duration is `10m`.
  google.protobuf.Duration mesh_edges_reporting_duration = 4;

  // maximum size of the peer metadata cache.
  // A long lived proxy that connects with many transient peers can build up a
  // large cache. To turn off the cache, set this field to a negative value.
  int32 max_peer_cache_size = 5;

  // Optional: Disable using host header as a fallback if destination service is
  // not available from the controlplane. Disable the fallback if the host
  // header originates outsides the mesh, like at ingress.
  bool disable_host_header_fallback = 6;
}
